packages/client/src/components/Question/Questions.tsx

```diff
@@ -422,35 +440,57 @@ const Questions: FunctionComponent<QuestionsProps> = ({
         const { answer } = q;
 
         // Skip unanswered questions or in case of Contact Outcome
-        if (answer === undefined || contactOutcome) {
+        if (!showAllQuestions && (answer === undefined || contactOutcome)) {
           return null;
         }
 
         // Get new index
         const index = i + 1 + checker.stack.length;
 
         // Check if current question is causing a outcome
-        const showQuestionAlert = !!permitsPerQuestion[index];
+        const showQuestionAlert = !!permitsPerQuestion[index] && !isForm;
 
         // Disable the EditButton or not
         const disabled = checker.isConclusive() || disableFutureQuestions;
 
         // Define the outcome type
-        const outcomeType: ClientOutcomes = permitsPerQuestion[i];
+        const outcomeType: imtr.ClientOutcomes = permitsPerQuestion[i];
+
+        // Check if this is the last question
+        const isFinalQuestion =
```

Comment:
In the file above we use
```
 // Check if this is the last question
        const isFinalQuestion =
          !checker._getUpcomingQuestions().length &&
          checker.stack.length === i + 1;
```
Is this correct?

Reply:
Do you mean if it's correct that they both use two different ways to determine the last question? That is correct. However, this code could be prettier indeed.